exonum/src/node/mod.rs

```diff
@@ -805,6 +809,38 @@ pub struct ConnectInfo {
     pub public_key: PublicKey,
 }
 
+impl ConnectInfo {
+    /// Tries to serialize the given `ConnectInfo` into vector of bytes.
+    pub fn try_serialize(self) -> Vec<u8> {
+        let mut vec_bytes = vec![0_u8; Self::field_size() as usize];
+        self.write(&mut vec_bytes, 0, Self::field_size());
+        vec_bytes
+    }
+
+    /// Tries to deserialize the given `value` into `ConnectInfo`.
+    #[allow(unsafe_code)]
+    pub fn try_deserialize(value: &[u8]) -> Self {
+        unsafe { Self::read(&value, 0, Self::field_size()) }
+    }
+}
```

Comment:
This looks suspicious.

The methods are named with `try` prefix, but based on the return types they do not enable error handling. Usually `try_somethings()` return `Option` or `Result`. That's in contrast with regular `something()` which is expected to panic on error condition. Furthermore, serialization consumes `self` which is a bit unusual. These functions seem to be used only in `impl StorageValue`, I guess we can just inline them there and do not provide this functionality separately.

Also, the `unsafe` block raises some questions. Is it really safe? For example, what if the `value` slice is too short or contains corrupted data?